app/routes/ffxiv.weekly-price-group-delta.tsx

```
@@ -0,0 +1,628 @@
+  if (actionData && 'data' in actionData) {
+    // Get all unique timestamps across all groups
+    const allTimestamps = Array.from(
+      new Set(
+        Object.values(actionData.data).flatMap((groupData) =>
+          Object.keys(groupData.deltas)
+        )
+      )
+    ).sort()
+
+    // Initialize selected dates to full range
+    useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)